File: docs/snippets/go/cli/cli_with_config.go
```go
package main

import (
	"encoding/json"
	"fmt"
	"os/exec"
)

type ExtractionResult struct {
	Content   string   `json:"content"`
	Format    string   `json:"format"`
	Languages []string `json:"languages"`
}

func extractWithConfig(filePath string, configPath string) (*ExtractionResult, error) {
	cmd := exec.Command(
		"kreuzberg",
		"extract",
		filePath,
		"--config",
		configPath,
		"--format",
		"json",
	)

	output, err := cmd.CombinedOutput()
	if err != nil {
		return nil, fmt.Errorf("CLI error: %w, output: %s", err, string(output))
	}

	var result ExtractionResult
	if err := json.Unmarshal(output, &result); err != nil {
		return nil, fmt.Errorf("JSON parse error: %w", err)
	}

	return &result, nil
}

func main() {
	configFile := "kreuzberg.toml"
	document := "document.pdf"

	fmt.Printf("Extracting %s with config %s\n", document, configFile)
	result, err := extractWithConfig(document, configFile)
	if err != nil {
		panic(err)
	}

	fmt.Printf("Content length: %d\n", len(result.Content))
	fmt.Printf("Format: %s\n", result.Format)
	fmt.Printf("Languages: %v\n", result.Languages)
}
```

File: docs/snippets/typescript/cli/cli_with_config.ts
```typescript
import { spawn } from "child_process";

interface ExtractionResult {
  content: string;
  format?: string;
  languages?: string[];
}

async function extractWithConfig(
  filePath: string,
  configPath: string
): Promise<ExtractionResult> {
  return new Promise((resolve, reject) => {
    const child = spawn("kreuzberg", [
      "extract",
      filePath,
      "--config",
      configPath,
      "--format",
      "json",
    ]);

    let stdout = "";
    let stderr = "";

    child.stdout.on("data", (data) => {
      stdout += data.toString();
    });

    child.stderr.on("data", (data) => {
      stderr += data.toString();
    });

    child.on("close", (code) => {
      if (code !== 0) {
        reject(new Error(`CLI exited with code ${code}: ${stderr}`));
        return;
      }

      resolve(JSON.parse(stdout));
    });
  });
}

const configFile = "kreuzberg.toml";
const document = "document.pdf";

console.log(`Extracting ${document} with config ${configFile}`);
const result = await extractWithConfig(document, configFile);

console.log(`Content length: ${result.content.length}`);
console.log(`Format: ${result.format}`);
console.log(`Languages: ${result.languages?.join(", ")}`);
```
